easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,157 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Xmipp, implemented as an easyblock
+
+@author: Jens Timmerman (Ghent University)
+"""
+import glob
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.filetools import mkdir, extract_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
+
+
+class EB_Xmipp(EasyBlock):
+    """Support for building/installing Xmipp."""
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, enable building in installation directory."""
+        super(EB_Xmipp, self).__init__(*args, **kwargs)
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Extract sources."""
+        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
+        self.cfg.update('unpack_options', '--strip-components=1')
+        super(EB_Xmipp, self).extract_step()
+
+    def configure_step(self):
+        """Configure xmipp build via a provided wrapper around sconss."""
+        # check if all our dependencies are in place
+        python_root = get_software_root('Python')
+        if not python_root:
+            self.log.error("Python not loaded as a dependency, which is required for %s" % self.name)
+        python_libdir = det_pylibdir()
+        self.python_short_ver = ".".join(get_software_version('Python').split(".")[0:2])
+        java_root = get_software_root('Python')
+        if not java_root:
+            self.log.error("Java not loaded as a dependency, which is required for %s" % self.name)
+
+        # extract some dependencies that we really need and can't find anywhere else
+        # alglib tarball has version in name, so lets find it with a glob
+        # we can't do this in extract step before these are in the original sources tarball, so we need to know
+        # startdir first
+        external_path = os.path.join(self.cfg['start_dir'], 'external')
+        alglib_tar = glob.glob(os.path.join(external_path, 'alglib*.tgz'))[0]
+        for src in ['bilib.tgz', 'bilib.tgz', 'condor.tgz', alglib_tar, 'scons.tgz']:
+            extract_file(os.path.join(external_path, src), external_path)
+
+        # make sure we start in the start dir
+        os.chdir(self.cfg['start_dird'])
+
+        # build step expects these to exist
+        mkdir(os.path.join(self.cfg['start_dir'], 'bin'))
+        mkdir(os.path.join(self.cfg['start_dir'], 'lib'))
+
+        cmd = ' '.join([
+            '{preconfigopts}',
+            'python external/scons/scons.py mode=configure -j {parallel}',
```

Comment:
these have nothing to do with each other, so split up in separate entries/lines?